engine/Assets/Scripts/UI/Dynamic/Modals/Configuring/ConfigMotorModal.cs

```diff
@@ -0,0 +1,261 @@
+                _turnDrivers[i]       = _robot.modules[i].azimuth;
+            }
+
+            motorCount -= _turnMotors.Length;
+        } else {
+            int wheelCount = _robot.GetLeftRightWheels()!.Value.leftWheels.Count + _robot.GetLeftRightWheels()!.Value.rightWheels.Count;
+            _driveMotors    = new ConfigMotor[wheelCount];
+            _driveDrivers   = new WheelDriver[wheelCount];
+            _robot.GetLeftRightWheels()!.Value.leftWheels.ForEach(x => {
+                _driveMotors[i]        = new ConfigMotor();
+                _driveMotors[i].driver = x;
+                _driveDrivers[i]       = x;
+                i++;
+            });
+            _robot.GetLeftRightWheels()!.Value.rightWheels.ForEach(x => {
+                _driveMotors[i]        = new ConfigMotor();
+                _driveMotors[i].driver = x;
+                _driveDrivers[i]       = x;
+                i++;
+            });
+        }
+        
+        //grab original target velocity for drivetrain
+        motorCount             += SimulationManager.Drivers[_robot.Name].Count - _driveMotors.Length;
+        _driveMotors[0].origVel = (_driveMotors[0].driver as WheelDriver).Motor.targetVelocity;
+        if (_robotISSwerve) {
+            _turnMotors[0].origVel = (_turnMotors[0].driver as RotationalDriver).Motor.targetVelocity;
+        }
+
+        Driver[] drivers = new Driver[SimulationManager.Drivers[_robot.Name].Count];
+        i = 0;
+        SimulationManager.Drivers[_robot.Name].ForEach(x => {
+            drivers[i] = x;
+        });
+
+        //set up other motors array
+        _otherMotors = new ConfigMotor[motorCount];
+        i = 0;
+        SimulationManager.Drivers[_robot.Name].ForEach(x => {
+            if (Array.IndexOf(_driveDrivers, x) == -1 && (!_robotISSwerve || Array.IndexOf(_turnDrivers, x) == -1)) {
+                _otherMotors[i]        = new ConfigMotor();
+                _otherMotors[i].driver = x;
+                i++;
+            }
+        });
+        
+        //UI
+        Title.SetText("Motor Configuration").SetWidth<Label>(MODAL_WIDTH - PADDING * 4);
+        Description.SetText("Change motor settings");
+
+        AcceptButton.StepIntoLabel(b => { b.SetText("Save"); }).AddOnClickedEvent(b => {
+            RobotSimObject.GetCurrentlyPossessedRobot().MiraLive.Save();
+            DynamicUIManager.CloseActiveModal();
+            
+            //Save to Mira
+            if (_driveMotors[0].velChanged) {
+                _driveMotors.ForEach(x => {
+                    SaveToMira(x.driver as WheelDriver);
+                });
+            }
+            if (_robotISSwerve && _turnMotors[0].velChanged) {
+                _turnMotors.ForEach(x => {
+                    SaveToMira(x.driver as RotationalDriver);
+                });
+            }
+            _otherMotors.ForEach(x => {
+                if (x.velChanged) {
+                    if (x.driver is WheelDriver) {
```

Comment:
There's a lot of checking and casting WheelDriver vs RotationalDriver even with the same code. Any ideas on how to simplify this? Wheel, Rotational, and Linear are all under Driver, though the motor config can only be done on Wheel and Rotational.

Reply:
Is it possible for `x.driver` to be `typeof()` something other than `WheelDriver` or `RotationalDriver`? Because if not we can let c# figure this out at runtime:

```cs
SaveToMira(x.driver as dynamic);
```

If `x.driver` can be of other types than this is probably the best solution with explicit casting.

Reply:
Yeah it can be a LinearDriver, which I'll work on once it's fixed in Fake Arm Control. If it's similar enough, I'll look into this. Good idea, thanks.